remove app engine gem, add runtime skip cleanup step

housseindjirdeh opened the pull request:
Apologies for having to have to submit another PR for this :(

The deploy step was unfortunately failing after you merged in my last PR (https://travis-ci.org/tastejs/hacker-news-pwas/builds/232316582). For some reason the `jekyll-app-engine` plugin wasn't created the `app.yaml` file correctly. I opened an issue for this [here](https://github.com/jamesramsay/jekyll-app-engine/issues/9). In the meantime however, I removed it and manually added the `app.yaml` file with the generated handlers. 

Tested this on a local repository and it seems to work (https://hnpwatest.appspot.com/).

A couple of things however:

1. If someone adds a static file/asset they'll have to update the `app.yaml` file accordingly. This isn't too big of a deal I think (and to be honest using wildcards will probably make more sense there, we can open a separate issue for that)
2. To set up a pipeline for App Engine with Travis CI, we need to make sure the service account file associated with the project is encrypted (which I did). Unfortunately, [**pull requests sent from forked repositories do not have access to encrypted data**](https://docs.travis-ci.com/user/pull-requests#Pull-Requests-and-Security-Restrictions). This means that anyone putting up a PR from a forked repo -- the build will fail at that script (meaning for every PR coming from a forked repo: the build will show that it fails). I can't think of a way around this unfortunately 🤔 . However, the following build once a PR is merged in should be accurate.

addyosmani commented:
@housseindjirdeh Thanks for investigating this. Subscribed to the issue you linked up to. I think, given the limitations we're aware of, this should be fine to move ahead with.

Do you think you might be able to throw in the extra notes at the bottom of the PR into a CONTRIBUTING.md so we know to watch out for it in the future?

> If someone adds a static file/asset they'll have to update the app.yaml file accordingly. This isn't too big of a deal I think (and to be honest using wildcards will probably make more sense there, we can open a separate issue for that)
> To set up a pipeline for App Engine with Travis CI, we need to make sure the service account file associated with the project is encrypted (which I did). Unfortunately, pull requests sent from forked repositories do not have access to encrypted data. This means that anyone putting up a PR from a forked repo -- the build will fail at that script (meaning for every PR coming from a forked repo: the build will show that it fails). I can't think of a way around this unfortunately 🤔 . However, the following build once a PR is merged in should be accurate.

housseindjirdeh commented:
Just added an environment variable to not run the decryption step for pull requests (https://github.com/tastejs/hacker-news-pwas/pull/28/files#diff-354f30a63fb0907d4ad57269548329e3R6). Since the `deploy` step won't happen for pull requests we should be okay! I didn't know we could check for pull requests when running a script step.

Nevertheless I added a `CONTRIBUTING.md` with some other points, please don't hesitate to change it up.
